riscv-executor/src/submachines.rs

```diff
@@ -44,152 +44,157 @@ pub trait Submachine<F: FieldElement> {
             last_row_overrides: cols.keys().map(|n| (n.clone(), None)).collect(),
             cols,
         }
     }
 
     fn len(&self) -> u32 {
+        if self.cols.is_empty() {
+            return 0;
+        }
         self.cols.values().next().unwrap().len().try_into().unwrap()
     }
 
     /// set the value of a column in all rows of the current block
-    fn set_current_block(&mut self, size: u32, col: &str, value: Elem<F>) {
+    fn set_current_block(&mut self, size: u32, col: &str, value: F) {
         for i in 0..size {
             let idx = self.len() - i - 1;
             *self
                 .cols
                 .get_mut(col)
-                .unwrap()
+                .unwrap_or_else(|| panic!("{} has no column {col}", self.namespace))
                 .get_mut(idx as usize)
                 .unwrap() = value;
         }
     }
 
     /// set the value of a column in the current
-    fn set_current_row(&mut self, col: &str, value: Elem<F>) {
-        *self.cols.get_mut(col).unwrap().last_mut().unwrap() = value;
+    fn set_current_row(&mut self, col: &str, value: F) {
+        *self
+            .cols
+            .get_mut(col)
+            .unwrap_or_else(|| panic!("{} has no column {col}", self.namespace))
+            .last_mut()
+            .unwrap() = value;
     }
 
     /// set the value of a column in the last row of the complete trace
-    fn set_final_row(&mut self, col: &str, value: Elem<F>) {
-        *self.last_row_overrides.get_mut(col).unwrap() = Some(value);
+    fn set_final_row(&mut self, col: &str, value: F) {
+        *self
+            .last_row_overrides
+            .get_mut(col)
+            .unwrap_or_else(|| panic!("{} has no column {col}", self.namespace)) = Some(value);
     }
 
     /// apply saved updates to the last row of the trace
     fn final_row_override(&mut self) {
         for (col, value) in self.last_row_overrides.iter() {
             if let Some(value) = value {
-                *self.cols.get_mut(col).unwrap().last_mut().unwrap() = *value;
+                *self
+                    .cols
+                    .get_mut(col)
+                    .unwrap_or_else(|| panic!("{} has no column {col}", self.namespace))
+                    .last_mut()
+                    .unwrap() = *value;
             }
         }
     }
 
     /// add new row of zeroes to the trce
     fn push_row(&mut self) {
-        self.cols
-            .values_mut()
-            .for_each(|v| v.push(Elem::Field(0.into())));
+        self.cols.values_mut().for_each(|v| v.push(0.into()));
     }
 
     /// Push a dummy block to the trace.
     /// A dummy block is a copy of the first block, with the final row updates applied to it, and selectors set to 0.
-    fn push_dummy_block(&mut self, machine_max_degree: usize, size: u32, selectors: &'static str) {
+    fn push_dummy_block(&mut self, machine_max_degree: u32, size: u32, selectors: &'static str) {
+        // TODO: use Optional selector argument
```

Comment:
here or later?

Reply:
removed the TODO changing the code here a bit